src/broker/terra-injection.js

```diff
@@ -0,0 +1,162 @@
+import { TerraNetworks } from '@liquality/terra-networks'
+
+import PortStream from 'extension-port-stream'
+import _ from 'lodash'
+import BN from 'bignumber.js'
+
+import { emitter } from '../store/utils'
+
+let isConnected = false
+let _address
+
+_.mixin({
+  toPairsDeep: obj => _.flatMap(
+    _.toPairs(obj),
+    ([k, v]) => _.isObjectLike(v) ? _.toPairsDeep(v) : [[k, v]]
+  )
+})
+
+const getValueForKey = (obj, key) => new Map(_.toPairsDeep(obj)).get(key)
+
+const getConfig = (activeNetwork) => {
+  const networkConfig = TerraNetworks['terra_' + activeNetwork]
+
+  const { chainID, nodeUrl: lcd, helperUrl: fcd, name } = networkConfig
+
+  return {
+    chainID,
+    fcd,
+    lcd,
+    name: activeNetwork === 'testnet' ? 'bombay' : name,
+    localterra: false
+  }
+}
+
+const getExecutedMethod = (msgs) => {
+  const parsed = msgs.map(msg => JSON.parse(msg))
+
+  const executeMessages = parsed.map(({ type, value }) => {
+    if (type === 'wasm/MsgExecuteContract') {
+      return value
+    }
+  })
+
+  if (!executeMessages.filter(Boolean).length) {
+    return 'send'
+  }
+
+  return Object.keys(executeMessages[executeMessages.length - 1]?.execute_msg)?.[0]
+}
+
+const getArgs = (payload) => {
+  const { fee, gasAdjustment, msgs } = payload
+
+  const { value } = JSON.parse(msgs[0])
+  const { amount, gas } = JSON.parse(fee)
+
+  const _value = getValueForKey(value, 'amount')
```

Comment:
It's kinda hard to be explicit because the structures of the messages are dynamic. Let me show you few examples:

Lock Collateral:
![image](https://user-images.githubusercontent.com/32274987/134778988-1e26cafe-ec85-4bd9-b66b-55987da5ada0.png)

Withdraw Collateral:
![image](https://user-images.githubusercontent.com/32274987/134778995-7a3762c6-4f11-467b-b0b8-d9d6259d9702.png)

Borrow Stable:
![image](https://user-images.githubusercontent.com/32274987/134779007-b8701be6-cf6e-4c14-9657-8526ada7e85d.png)

Repay Stable:
![image](https://user-images.githubusercontent.com/32274987/134779026-296b77a9-299d-408b-8722-8be354d093dd.png)


Mint:
![image](https://user-images.githubusercontent.com/32274987/134779034-2d40a90e-4608-4a8b-8d0a-5b8187775553.png)


Swap:
![image](https://user-images.githubusercontent.com/32274987/134779060-90caf16d-bef5-4cb8-9a5f-6e724b241478.png)

And few more different structures for the rest of the oprations

Reply:
Done :)